libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Login/LoginHost/SFSDKLoginHostListViewController.m

```diff
@@ -0,0 +1,284 @@
+}
+
+- (void)addLoginHost:(SFSDKLoginHost*)host {
+    [[SFSDKLoginHostStorage sharedInstance] addLoginHost:host];
+    
+    NSUInteger hostIndex = [[SFSDKLoginHostStorage sharedInstance] indexOfLoginHost:host];
+    if (hostIndex != NSNotFound) {
+        // Apply the selected login host
+        [self applyLoginHostAtIndex:hostIndex];
+        
+        // Notify the delegate that a new login host has been added.
+        [self delegateDidAddLoginHost];
+    }
+}
+
+- (void)showAddLoginHost {
+    [self showAddLoginHost:nil];
+}
+
+/**
+ * Invoked when the user presses the Add button. This method presents the new login host view.
+ */
+- (void)showAddLoginHost:(id)sender {
+    SFSDKNewLoginHostViewController *detailViewController = [[SFSDKNewLoginHostViewController alloc] initWithStyle:UITableViewStyleGrouped];
+    detailViewController.loginHostListViewController = self;
+    if (self.navigationController) {
+        self.navigationController.delegate = self;
+        [self.navigationController pushViewController:detailViewController animated:YES];
+    } else {
+        [self presentViewController:detailViewController animated:YES completion:nil];
+    }
+}
+
+#pragma mark - View lifecycle
+
+/**
+ * Set the proper size of the view so its popover controller will resize to fit neatly.
+ */
+- (void)resizeContentForPopover {
+    CGRect r = [self.tableView rectForSection:0];
+    
+    CGSize size = CGSizeMake(380, r.size.height);
+    self.preferredContentSize = size;
+}
+
+- (void)viewDidLoad {
+    self.navigationItem.rightBarButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemAdd target:self action:@selector(showAddLoginHost:)];
+    self.title = [SFSDKResourceUtils localizedString:@"LOGIN_CHOOSE_SERVER"];
+    
+    self.navigationItem.backBarButtonItem = [[UIBarButtonItem alloc]
+                                             initWithTitle:@""
+                                             style:UIBarButtonItemStylePlain
+                                             target:nil
+                                             action:nil];
+    self.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemCancel target:self action:@selector(cancelLoginPicker:)];
+    
+    // Make sure the current login host exists.
+    NSUInteger index = [self indexOfCurrentLoginHost];
+    if (NSNotFound == index) {
+        index = 0; // revert to standard in case there is no current login host
+        [self applyLoginHostAtIndex:index];
+    }
+    
+    // Refresh the UI and make sure the size is correct
+    [self.tableView reloadData];
+    [self makeLoginHostVisibleAtIndex:index];
+    [self resizeContentForPopover];
+    
+    [super viewDidLoad];
+    
+    [self.navigationItem.rightBarButtonItem setTintColor:[UIColor whiteColor]];
+    [self setEdgesForExtendedLayout:UIRectEdgeNone];
+    
+    //TODO: Remove the check once we move to iOS 9 as minimum.
```

Comment:
Added TODO here to remove the check once we move to iOS9 as minimum supported version.